tasks/checkInsolvencyAtMaturity.ts

```diff
@@ -6,28 +6,24 @@ import {
 } from "../typechain";
 import { ethers, utils, BigNumber } from "ethers";
 
-import { getPositions, Position } from "../scripts/getPositions";
 import { SECONDS_PER_YEAR } from "@aave/protocol-js";
 import { toBn } from "../test/helpers/toBn";
 
+import * as poolAddresses from "../pool-addresses/mainnet.json";
+
 import Decimal from "decimal.js-light";
+import { getPositions, Position } from "../scripts/getPositions";
 
 task(
   "checkInsolvencyAtMaturity",
   "Checks the insolvency status of positions at maturity by estimating the expected cashflow."
 ).setAction(async (taskArgs, hre) => {
-  // deploy FixedAndVariableMath as contract to be able to call function
-  const fixedAndVariableMathFactory = await hre.ethers.getContractFactory(
-    "FixedAndVariableMathTest"
-  );
-  const fixedAndVariableMath =
-    (await fixedAndVariableMathFactory.deploy()) as FixedAndVariableMathTest;
+  const fixedAndVariableMath = (await hre.ethers.getContractAt(
+    "FixedAndVariableMathTest",
+    "0x2D2EE238Ca74B546BfA64864f5654b5Ed7673f87"
```

Comment:
I guess this still only works with one network? If we fell back to deploying a new FixedAndVariableMathTest as before, would that be enough to save gas on mainnet but to make this task portable across networks?

Reply:
Good idea, thanks